regridding/regrid.py

```diff
@@ -497,26 +493,51 @@ def regrid_dataset(fp, regridder, out_fp, lat_slice):
 
     results = []
     errs = []
+    no_clobbers = []
+
     for fp in src_fps:
-        try:
-            out_fp = generate_regrid_filepath(fp, out_dir)
-            # make sure the parent dirs exist
-            out_fp.parent.mkdir(exist_ok=True, parents=True)
-            results.append(regrid_dataset(fp, regridder, out_fp, ext_lat_slice))
-        except Exception as e:
-            errs.append(str(fp))
+
+        out_fp = generate_regrid_filepath(fp, out_dir)
+        # make sure the parent dirs exist
+        out_fp.parent.mkdir(exist_ok=True, parents=True)
+
+        # list all existing files in the out_dir
+        existing_fps = list(out_dir.glob("**/*.nc"))
+        # remove date from filename about to be regridded
+        nodate_out_fp = "_".join(out_fp.name.split(".nc")[0].split("_")[:-1])
+
+        # search existing filenames for any beginning with the dateless string
+        # if any are found, and no_clobber=True, assume that all yearly files also exist and skip regridding
+        if (
+            any(str(fp.name).startswith(nodate_out_fp) for fp in existing_fps)
+            and no_clobber
+        ):
+            no_clobbers.append(str(fp))
             print(f"\nFILE NOT REGRIDDED: {fp}\n     Errors printed below:\n")
```

Comment:
I'd like to keep these messages, because the error for the specific file is printed. If we don't print it in the slurm output, those errors are lost and you'll never know why a file was not regridded. I think this will make for easier QC of any failures.